### DISCUSSION TITLE
Unknown Name Appearing On Commits Of My Repository

### DISCUSSION THREAD

**Discussion by mujibsayyad** [community/community#58240] [Repositories] [answered]
### Select Topic Area

Bug

### Feature Area

Issues

### Body

I encountered an issue where, after pushing code to my repository using my GitHub account, I noticed that the commits displayed someone else's name instead of mine. I have no knowledge of this individual and did not authorise them to make changes or commit code under my account. This situation raises concerns about the security and integrity of my repository. I am seeking assistance in resolving this issue and ensuring that my commits accurately reflect my identity.

i used command `git config --list` to check and it is showing me my right credentials 

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by Gustavonn07** [comment, marked as answer]
Hey how's it going? this problem can occur if you use a shared computer (at a college for example) this happens because the person has the credentials saved on the machine you used.

When this occurs, your commits end up becoming that person's commits in your repository, you can correct this by going to credentials and removing the person's github access, then logging in with your github in the terminal, type:
1. `git config --global user.name (your username)`
2. `git config --global user.email (your email)`


If it didn't work that way, see if you added that person as a contributor (that way he has access to change some defined files)

**Reply by NextGenGk** [reply]
Same problem in my account also 

How to resolve this issue

Please help!!

**Comment by mujibsayyad** [comment]
Hi,
It's my personal computer so no one uses except me.

So now i  re-updated my name and email
I pushed new changes
now its showing my name on commit.

Not sure what was the issue
i don't know who is that guy and  neither i gave any permission to him nor added him as contributor
still was a issue
but now it fixed.

I appreciate you help,
Thank You!

**Reply by Gustavonn07** [reply]
You wellcome!!

**Reply by mujibsayyad** [reply]
😀

**Comment by NextGenGk** [comment]
> Hi,
> It's my personal computer so no one uses except me.
> 
> So now i  re-updated my name and email
> I pushed new changes
> now its showing my name on commit.
> 
> Not sure what was the issue
> i don't know who is that guy and  neither i gave any permission to him nor added him as contributor
> still was a issue
> but now it fixed.
> 
> I appreciate you help,
> Thank You!
> 

Hey man!! 
How did you fix this problem?
Please help i'm also stuck in this issue

**Reply by mujibsayyad** [reply]
Hey,
Open terminal and follow commands

git config --global user.name yourgithubusername
git config --global user.email your@email.com

then try pushing new changes it will show your name on latest changes
one thing to notice, it will not change name on previous commit so the unknown users name will persist there until you remove that commit

**Reply by NextGenGk** [reply]
Bro My contribution is not shown in my profile? 
When I am push a changes in our github repo contribution is not count in my profile 
Can you please help me

**Reply by NextGenGk** [reply]
> Hey,
> Open terminal and follow commands
> 
> git config --global user.name yourgithubusername
> git config --global user.email your@email.com
> 
> then try pushing new changes it will show your name on latest changes
> one thing to notice, it will not change name on previous commit so the unknown users name will persist there until you remove that commit
> 

Ok i will check

**Comment by NextGenGk** [comment]
Thnk u @mujibsayyad @Gustavonn07

**Comment by king-ellie** [comment]
Hey everyone, I just encountered this on a project. I understand the fix, but does anyone understand _why_ this happened? This is absolutely bizarre behavior

**Comment by darkspywarrior** [comment]
Hey everyone!
I’ve been facing an issue where my commits show an unknown person’s name, even though my global and local Git configs have my correct credentials. I’m not using a shared computer. When I push from terminal, it shows the wrong name, but uploading files directly shows my name correctly. I’d really appreciate help to resolve this quickly!

![Screenshot 2025-06-26 115936](https://github.com/user-attachments/assets/3c316a9c-62cc-49c4-a18a-e08ee8eada23)

**Reply by zyreyx** [reply]
Hello! Did you ever resolve this issue? The same thing is happening to me and from the same account

**Comment by darkspywarrior** [comment]
try changing global username and global email id or try author one git
commit --amend --author="Your Correct Name ". someone once told me they
deleted smtg everything just got right.
Hope everything works out
regards
anika
